cloud/scope/cluster.go

```diff
@@ -383,6 +383,144 @@ func (s *ClusterScope) detachPublicGateway(subnetID string, pgwID string) error
 	return err
 }
 
+// CreateLoadBalancer creates a new IBM VPC load balancer in specified resource group.
+func (s *ClusterScope) CreateLoadBalancer() (*vpcv1.LoadBalancer, error) {
+	loadBalancerReply, err := s.ensureLoadBalancerUnique(s.IBMVPCCluster.Spec.ControlPlaneLoadBalancer.Name)
+	if err != nil {
+		return nil, err
+	} else if loadBalancerReply != nil {
+		// TODO need a reasonable wrapped error
+		return loadBalancerReply, nil
+	}
+
+	options := &vpcv1.CreateLoadBalancerOptions{}
+	options.SetName(s.IBMVPCCluster.Spec.ControlPlaneLoadBalancer.Name)
+	options.SetIsPublic(true)
+	options.SetResourceGroup(&vpcv1.ResourceGroupIdentity{
+		ID: &s.IBMVPCCluster.Spec.ResourceGroup,
+	})
+
+	if s.IBMVPCCluster.Status.Subnet.ID != nil {
+		options.Subnets = []vpcv1.SubnetIdentityIntf{}
+		subnet := &vpcv1.SubnetIdentity{
+			ID: s.IBMVPCCluster.Status.Subnet.ID,
+		}
+		options.Subnets = append(options.Subnets, subnet)
+	}
+
+	options.SetPools([]vpcv1.LoadBalancerPoolPrototype{
+		{
+			Algorithm:     core.StringPtr("round_robin"),
+			HealthMonitor: &vpcv1.LoadBalancerPoolHealthMonitorPrototype{Delay: core.Int64Ptr(5), MaxRetries: core.Int64Ptr(2), Timeout: core.Int64Ptr(2), Type: core.StringPtr("tcp")},
+			Name:          core.StringPtr(s.IBMVPCCluster.Spec.ControlPlaneLoadBalancer.Name + "-pool"),
+			Protocol:      core.StringPtr("tcp"),
+		},
+	})
+
+	options.SetListeners([]vpcv1.LoadBalancerListenerPrototypeLoadBalancerContext{
+		{
+			Protocol: core.StringPtr("tcp"),
+			Port:     core.Int64Ptr(6443),
+			DefaultPool: &vpcv1.LoadBalancerPoolIdentityByName{
+				Name: core.StringPtr(s.IBMVPCCluster.Spec.ControlPlaneLoadBalancer.Name + "-pool"),
+			},
+		},
+		{
+			Protocol: core.StringPtr("tcp"),
+			Port:     core.Int64Ptr(22),
```

Comment:
To enable ssh connectivity with control-plane node, normally this should not be required. Should I go ahead and drop the same?

Reply:
As this is load balanced to multiple backend IP addresses, there is no guarantee that it will goto the expected machine always, it better to remove this entry.. adding ssh is out os scope for this work, will discuss on that in detail how to deal with that..

Reply:
Fixed.